commit 04521ed5f795695185a0e9bbed3b91d2e47ba540 (refs/changes/35/2735/1)
Author: Kelly McLaughlin <kelly.mclaughlin@joyent.com>
Date:   2017-10-06T09:25:16-06:00 (2 years ago)
    
    MANATEE-323 manatee misguidedly convinced that if it calls watch() on a dead ZK client enough, it will come back to life

diff --git a/lib/zookeeperMgr.js b/lib/zookeeperMgr.js
index 7fcfba5..55ef126 100644
--- a/lib/zookeeperMgr.js
+++ b/lib/zookeeperMgr.js
@@ -247,7 +247,9 @@ function watch(zk, getFunc, path, wat, cb) {
                 path: path,
                 err: err
             }, 'zk: error fetching zk data or children');
-            setTimeout(registerWatch, 5000);
+            if (err.code !== zkClient.Exception.CONNECTION_LOSS) {
+                setTimeout(registerWatch, 5000);
+            }
         } else {
             wat.call(self, watRes);
         }
